kubernetes/charts/weblogic-operator/values.yaml

```diff
@@ -159,6 +159,20 @@ javaLoggingFileCount: 10
 # the operator's deployment. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
 # for more information on node selectors.
 #nodeSelector:
+ 
+# webhookOnly specifies whether only the webhook deployment named weblogic-operator-webhook
+# should be created during helm install and if the creation of the operator deployment should be skipped.
+# By default, the helm install will create both the webhook deployment and the operator deployment.
+# If set to true, the helm install will only create the webhook deployment.
+# The default value is false.
+webhookOnly: false
+ 
+# preserveWebhook specifies whether the previous webhook deployment should be preserved
+# when the chart is uninstalled using helm uninstall.
+# By default, the helm uninstall will remove both the webhook deployment and the operator deployment.
+# If set to true, the helm uninstall command will skip removing the webhook deployment.
+# The default value is false.
+preserveWebhook: false
```

Comment:
The CRD behavior is still the same and operator uninstall would leave the CRD behind. The `preserveWebhook` option is if the webhook pod(s) needs to be left behind. I think majority of use-cases will have single operator and they will be surprised to see a running pod after they uninstall the operator. For multiple operators use-cases, we can recommend customers to install the webhook in a namespace that is separate from the operator namespace (using `webhookOnly` option). This way webhook will not be uninstalled when the operator is uninstalled. I would vote to keep this default to `false`.

Reply:
I will leave it up to you. I think it's more intuitive and safer to have it match the CRD life-cycle.  

But if you do leave it 'false', please make absolutely sure the documentation for operator cleanup (in quickstart, etc, etc) is extremely clear about the potential "silent" side effects... 

To me, this is like having a light switch in every office that surprisingly can shut down the entire building unless you jiggle it in exactly the right way when switching it off.  Works fine in a single office building, but not otherwise.

Reply:
ok. I believe we discussed this previously and my understanding is that we decided to keep it simple for ~ 90% of the customers who have a single operator. I agree on the documentation part. I'll leave the default to 'false' unless @rjeberhard and @robertpatrick disagrees and we can change the default later if we decide to do it.